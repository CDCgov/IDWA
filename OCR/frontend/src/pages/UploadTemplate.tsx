import { useNavigate } from "react-router-dom"
import { Divider } from "../componets/Divider"
import { UploadHeader } from "../componets/Header"
import { Stepper } from "../componets/Stepper"
import { Uploadfile } from "../componets/UploadFile"
import { AnnotateStep } from "../utils/constants"

export const UploadTemplate = () => {
    const navigate = useNavigate();
    return (
        <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'start', width: '100%', height: '100%', padding: '16px 8px, 8px, 8px' }}>
            <UploadHeader title="Annotate new template" onBack={() => {}} onSubmit={() => navigate('/new-template/annotate')} />
            <Divider margin="0px" />
            <div style={{ display: 'flex', justifyContent: 'center', paddingTop: '32px'}}>
                <Stepper currentStep={AnnotateStep.Upload} />
            </div>
            <Divider margin="0px" />
            <Uploadfile />
        </div>
    )
}