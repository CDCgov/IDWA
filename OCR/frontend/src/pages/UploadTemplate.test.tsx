import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { UploadTemplate } from './UploadTemplate';

describe('UploadTemplate component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<UploadTemplate />);
    expect(asFragment()).toMatchSnapshot();
  });
});
