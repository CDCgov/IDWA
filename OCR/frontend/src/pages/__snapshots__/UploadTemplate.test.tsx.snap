// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UploadTemplate component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; justify-content: start; width: 100%; height: 100%;"
  >
    <header
      class="usa-header"
      data-testid="header"
      style="height: 50px; padding: 8px;"
    >
      <div
        class="display-flex height-full flex-align-center"
        style="justify-content: space-between;"
      >
        <div
          class="display-flex flex-jusitfy-start height-full flex-align-center"
          style="justify-content: space-between;"
        >
          <button
            class="usa-button usa-button--unstyled"
            data-testid="close-button"
            style="padding-right: 8px;"
            type="button"
          >
            <svg
              class="usa-icon usa-icon--size-3"
              color="black"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
          <h1>
            Annotate new template
          </h1>
        </div>
        <div
          class="display-flex flex-jusitfy-start height-full flex-align-center"
        >
          <button
            class="usa-button usa-button--outline"
            data-testid="button"
            style="height: 40px; box-shadow: inset 0 0 0 2px  #adadad; color: rgb(173, 173, 173);"
            type="reset"
          >
            Back
          </button>
          <button
            class="usa-button usa-button--base"
            data-testid="button"
            style="height: 40px;"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </header>
    <div
      style="border-bottom: 1px solid #ccc; margin: 0px;"
    />
    <div
      style="display: flex; justify-content: center; padding-top: 32px;"
    >
      <div
        aria-label="progress"
        class="usa-step-indicator"
        data-testid="step-indicator"
      >
        <ol
          class="usa-step-indicator__segments"
        >
          <li
            aria-current="true"
            class="usa-step-indicator__segment usa-step-indicator__segment--current"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Upload New Segment 
            </span>
          </li>
          <li
            class="usa-step-indicator__segment"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Annotate 
              <span
                class="usa-sr-only"
              >
                not completed
              </span>
            </span>
          </li>
          <li
            class="usa-step-indicator__segment"
          >
            <span
              class="usa-step-indicator__segment-label"
            >
              Save Templete 
              <span
                class="usa-sr-only"
              >
                not completed
              </span>
            </span>
          </li>
        </ol>
        <div
          class="usa-step-indicator__header"
        >
          <h5
            class="usa-step-indicator__heading"
            style="display: none;"
          >
            <span
              class="usa-step-indicator__heading-counter"
            >
              <span
                class="usa-sr-only"
                data-testid="step-text"
              >
                Step
              </span>
              <span
                class="usa-step-indicator__current-step"
              >
                1
              </span>
               
              <span
                class="usa-step-indicator__total-steps"
              >
                of 3
              </span>
               
            </span>
            <span
              class="usa-step-indicator__heading-text"
            >
              Upload New Segment
            </span>
          </h5>
        </div>
      </div>
    </div>
    <div
      style="border-bottom: 1px solid #ccc; margin: 0px;"
    />
    <div
      class="display-flex flex-column flex-align-center flex-justify-start height-full width-full padding-2 bg-primary-lighter"
    >
      <div
        style="width: 70%; text-align: left;"
      >
        <h1
          style="margin: 0px;"
        >
          Upload new form to segment
        </h1>
        <h2
          style="margin: 0px;"
        >
          Upload new image or PDF and save as a template
        </h2>
      </div>
      <div
        class="display-flex flex-column margin-top-205 flex-justify flex-align-center bg-white"
        data-testid="dashed-container"
        style="width: 70%; height: 40%; border: 1px dashed #005ea2;"
      >
        <svg
          class="usa-icon usa-icon--size-5"
          color="#005ea2"
          data-testid="upload-icon"
          focusable="false"
          height="1em"
          role="img"
          style="margin-top: 16px;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11zM8 15.01l1.41 1.41L11 14.84V19h2v-4.16l1.59 1.59L16 15.01 12.01 11z"
          />
        </svg>
        <div
          class="display-flex flex-column flex-align-center margin-bottom-1"
          style="width: 60%;"
        >
          <h3
            style="font-weight: bold;"
          >
            Drag and drop file here
          </h3>
          <p>
            or
          </p>
          <div
            class="usa-file-input padding-bottom-2"
            data-testid="file-input"
          >
            <div
              class="usa-file-input__target"
              data-testid="file-input-droptarget"
            >
              <div
                aria-hidden="true"
                class="usa-file-input__instructions"
                data-testid="file-input-instructions"
              >
                <span
                  class="usa-file-input__drag-text"
                >
                    
                </span>
                <span
                  class="usa-file-input__choose"
                >
                  Browse Files
                </span>
              </div>
              <div
                class="usa-file-input__box"
                data-testid="file-input-box"
              />
              <input
                class="usa-file-input__input"
                data-testid="file-input-input"
                id="file-input-:r0:"
                name="file-input-single"
                style="border: 1px dashed #005ea2;"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
